### ISSUE TITLE
Introduce a way to add custom colours to COLOR_MAP easily.

### ISSUE THREAD

**Issue by Aathish04** [ManimCommunity/manim#279]
It would be extremely useful to the end user if they could add their own colours to the COLOR_MAP/PALETTE without having to modify `constants.py`.

I'm thinking a section in the cfg files could be added with the entry `CUSTOM_COLORS` or similar, and the dictionary corresponding to it could be merged with COLOR_MAP at runtime?

**Comment by leotrs** [comment]
I like your suggestion!

**Comment by leotrs** [comment]
Related to #310

**Comment by huguesdevimeux** [comment]
Is there a way to do this with the Enum implemented in #342 ?

**Comment by leotrs** [comment]
I'm not sure that Enum should be implemented at all. I think we should discuss/implement/close #342 before coming back and reconsidering this one .

**Comment by kolibril13** [comment]
In matplotlib one can set default colourmaps like this:
```
import matplotlib.pyplot as plt
plt.set_cmap('coolwarm')
```
can we maybe have a similar approach?
So the idea is:
all colours already have their variables, like WHITE, BLACK, YELLOW etc.
which are called like this in
WHITE = Colors.white.value
colors.py, 
but what actually *is* the colours is looked up by what colormap is currently active.

**Comment by leotrs** [comment]
> but what actually is the colours is looked up by what colormap is currently active.

I don't understand what you mean by this.

Also, before implementing anything more on colors, I'd like to finish and close #342. It's stuck on needing more feedback because the proposed implementation doesn't seem to fix all existing problems.

**Comment by leotrs** [comment]
This was being blocked by #342, which was actually superseded by #488, which has been merged. So let's pick this one back up.

@kolibril13 I think I understand you now! Is this what you mean?
```python
>>> import manim as mn
>>> from manim import color as C 
>>> C.RED
'#FC6255'
>>> mn.set_cmap("logo")
>>> C.RED
'#E07A5A'    # <-- our logo's red
```

**Comment by kolibril13** [comment]
Yess, exactly!

**Comment by leotrs** [comment]
I think I'm onboard with this in general. However, note there will be a big difference between matplotlib and manim. When you change colormap in matplotlib, it changes the default color that it uses _when you don't specify a color_. For example, the color of the line plotted by `plt.plot(x, y)` changes. But even after choosing a colormap, you can do `plt.plot(x, y, c='r')` and it will use the standard red.

The example above actually changes what `C.RED` means. Which is distinctly different to the above. So I'm not entirely sold yet. Would like to see what others think.

**Comment by jsonvillanueva** [comment]
#784 I believe this plugin feature could handle this issue as soon as it's merged in. One could create a plugin, `manim-ThemeManager` which could be responsible for handling color maps (i.e. this doesn't have to be part of core manim). How the author of such a plugin wishes to treat this is entirely up to them, but a suggestion would be to have predefined color palettes in a convenient format like json. 

```python
>>> from manim import *
>>> from manim_thememanager import tm
>>> tm.set_style("DarkTheme.json")
>>> tm.set_highlights("OtherTheme.json")
>>> # Do rest of project code using custom color
>>> Circle(color=tm.RED) # or something similar
```

**Comment by leotrs** [comment]
I think I like that idea. This allows for a clean core and complete freedom to determine how color themes work.
